src/Component/RegistrationForm.js

```diff
@@ -0,0 +1,278 @@
+import React, { Component } from 'react'
+import '../Stylesheet/RegistrationForm.css'
+
+class RegistrationForm extends Component {
+  constructor(props) {
+    super(props)
+    this.isInputsValid = true
+    this.state = {
+      loginId: {
+        value:'',
+        error:''
+      },
+      email: {
+        value:'',
+        error:''
+      },
+      name: {
+        value:'',
+        error:''
+      },
+      url: {
+        value:'',
+        error:''
+      },
+      timezone: {
+        value:'SelectTimezone',
+        error:''
+      },
+      aboutMe: {
+        value:'',
+        error:''
+      },
+      receiveNotification: false,
+      showCheckboxError: false,
+    }
+  }
+  checkEmailValidation(email) {
+    const regex = (/^(([^<>()[\]\.,;:\s@]+(\.[^<>()[\]\.,;:\s@]+)*)|(.+))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/)
+    return regex.test(email);
+  }
+  checkUrlValidation(url) {
+    const regex = (/^(http[s]?:\/\/(www\.)?|ftp:\/\/(www\.)?|www\.){1}([0-9A-Za-z-\.@:%_\+~#=]+)+((\.[a-zA-Z]{2,3})+)((.)*)?(\?(.)*)?/)
+    return regex.test(url);
+  }
+  handleFormSubmission = (e) => {
+    console.log('On submit called')
+    this.isInputsValid = true
+    this.checkLogin()
```

Comment:
all of these methods look repetitive. Write a method that takes the field as arg and call helpers method calling to reduce code.

Reply:
Create a method with field , value and error arguments will set state for the variable.